nautobot/extras/context_managers.py

```diff
@@ -16,14 +17,14 @@ class ChangeContext:
     one will be generated to relate any changes to this transaction. Convenience
     classes are provided for each context.
 
-    :param user: User object
+    :param request: WSGIRequest object
```

Comment:
Should this change be specific to the `WebChangeContext` class? It feels strange to pass a `request` when creating a `JobHookChangeContext` for example. Wasn't part of the goal of creating the various `ChangeContext` classes to move *away* from a close linkage to HTTP requests?

Reply:
Agreed. I added support in the base class for both the `request` and `user` kwargs and only use `request` where necessary to support DRF.